lib/filewatch/helper.rb

```diff
@@ -0,0 +1,73 @@
+# code downloaded from Ruby on Rails 4.2.1
+# https://raw.githubusercontent.com/rails/rails/v4.2.1/activesupport/lib/active_support/core_ext/file/atomic.rb
+require 'fileutils'
+
+class File
+  # Write to a file atomically. Useful for situations where you don't
+  # want other processes or threads to see half-written files.
+  #
+  #   File.atomic_write('important.file') do |file|
+  #     file.write('hello')
+  #   end
+  #
+  # If your temp directory is not on the same filesystem as the file you're
+  # trying to write, you can provide a different temporary directory.
+  #
+  #   File.atomic_write('/data/something.important', '/data/tmp') do |file|
+  #     file.write('hello')
+  #   end
+  def self.atomic_write(file_name, temp_dir = Dir.tmpdir)
+    require 'tempfile' unless defined?(Tempfile)
+    require 'fileutils' unless defined?(FileUtils)
+
+    temp_file = Tempfile.new(basename(file_name), temp_dir)
```

Comment:
Recommend not using Tempfile, since it has some odd behaviors (close/delete on GC finalizer, delete on close?). I'm OK with Tempfile if this isn't a problem in this scenario.

Stud::Temporary exists in cases where Tempfile isn't a good friend.